src/lib.rs

```diff
@@ -272,6 +275,33 @@ impl StreamDeck {
         self.write_button_image(key, &image)
     }
 
+    /// Sets a button to the provided text.
+    /// Will break text over \n linebreaks
+    pub fn set_button_text(
+        &mut self,
+        key: u8,
+        colour: &Colour,
+        background: &Colour,
+        x: u32,
+        y: u32,
+        scale: Scale,
+        line_height: f32,
+        font: &Font,
+        text: &str,
+    ) -> Result<(), Error> {
+        let (width, height) = self.kind.image_size();
+        let background = Rgb([background.r, background.g, background.b]);
+        let colour = Rgb([colour.r, colour.g, colour.b]);
+        let mut image = ImageBuffer::from_pixel(width as u32, height as u32, background);
+        let mut y = y;
+        text.to_string().split("\n").for_each(|txt| {
+            draw_text_mut(&mut image, colour, x, y, scale, font, txt);
```

Comment:
i wonder if is it possible to automatically centre (or more generally align, though i think centring would be enough) this in a useful way so that people don't have to deal with the x/y offsets for a given piece of text?

Reply:
I can't work out how to find out the width of the text :cry:

Reply:
Hmm, doesn't seem super easy! I _think_ it should be possible using something like [`Font::layout`](https://docs.rs/rusttype/0.9.2/rusttype/enum.Font.html#method.layout) or the via looking at the glyphs manually.

To avoid this for now we can put a `Position` enum in the `ImageOptions` object with an `Absolute{x: .., y: ..}` variant so adding this later just involves an additional variant and is thus a non-breaking change.

Reply:
That's a fantastic idea. :+1: